File: kafka-consumer/src/main/java/net/opentsdb/kafka/consumer/modules/ConsumerModule.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package net.opentsdb.kafka.consumer.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.google.inject.name.Names;

import java.util.Properties;

/**
 * Guice Module used to bind instances
 */
public class ConsumerModule extends AbstractModule {
  private final Properties properties;

  public ConsumerModule(Properties properties) {
    this.properties = properties;
  }

  @Override
  protected void configure() {
    Names.bindProperties(binder(), properties);
  }

  @Provides
  @Singleton
  Properties provideProperties() {
    return properties;
  }
}

File: kafka-consumer/src/main/scala/net/opentsdb/kafka/consumer/modules/ConsumerModule.scala
package net.opentsdb.kafka.consumer.modules

import com.google.inject.{Singleton, Provides, AbstractModule}
import net.codingwell.scalaguice.ScalaModule
import java.util.Properties
import com.google.inject.name.{Named, Names}
import java.util.concurrent.{LinkedBlockingQueue, ArrayBlockingQueue}
import org.hbase.async.HBaseClient
import net.opentsdb.core.TSDB

class ConsumerModule(props: Properties) extends AbstractModule with ScalaModule {
  def configure() {
    Names.bindProperties(binder(), props)
  }

  @Provides @Singleton def provideProperties: Properties = props

  @Provides @Singleton def provideHBaseClient(@Named("hbase.zookeeper") zk: String): HBaseClient = {
    val client = new HBaseClient(zk)

    client.setFlushInterval(500)

    client
  }

  @Provides @Singleton def provideTsdbClient(@Named("tsdb.table") tsdbTable: String, @Named("tsdb.uidtable") uidTable: String,
    client: HBaseClient): TSDB = new TSDB(client, tsdbTable, uidTable)
}
